rust/agama-cli/src/logs.rs

```diff
@@ -181,64 +218,77 @@ impl LogItem for LogCmd {
     }
 }
 
-// collect existing / requested paths which should already exist turns them into list of log
-// sources
-fn paths_to_log_sources(paths: &[&str], tmp_dir: &TempDir) -> Vec<Box<dyn LogItem>> {
+// Collect existing / requested paths which should already exist in the system.
+// Turns them into list of log sources
+fn paths_to_log_sources(paths: &Vec<String>, tmp_dir: &TempDir) -> Vec<Box<dyn LogItem>> {
     let mut log_sources: Vec<Box<dyn LogItem>> = Vec::new();
 
-    for path in paths {
+    for path in paths.iter() {
         // assumption: path is full path
         if Path::new(path).try_exists().is_ok() {
-            log_sources.push(Box::new(LogPath::new(path, tmp_dir.path())));
+            log_sources.push(Box::new(LogPath::new(path.as_str(), tmp_dir.path())));
         }
     }
 
     log_sources
 }
 
-// some info can be collected via particular commands only, turn it into log sources
-fn cmds_to_log_sources(commands: &[&str], tmp_dir: &TempDir) -> Vec<Box<dyn LogItem>> {
+// Some info can be collected via particular commands only, turn it into log sources
+fn cmds_to_log_sources(commands: &Vec<String>, tmp_dir: &TempDir) -> Vec<Box<dyn LogItem>> {
     let mut log_sources: Vec<Box<dyn LogItem>> = Vec::new();
 
-    for cmd in commands {
-        log_sources.push(Box::new(LogCmd::new(cmd, tmp_dir.path())));
+    for cmd in commands.iter() {
+        log_sources.push(Box::new(LogCmd::new(cmd.as_str(), tmp_dir.path())));
     }
 
     log_sources
 }
 
-// compress given directory into a tar archive
+// Compress given directory into a tar archive
 fn compress_logs(tmp_dir: &TempDir, result: &String) -> io::Result<()> {
     let compression = DEFAULT_COMPRESSION.0;
+    let tmp_path = tmp_dir
+        .path()
+        .parent()
+        .unwrap()
+        .as_os_str()
+        .to_str()
+        .ok_or(io::Error::new(io::ErrorKind::InvalidInput, "Wrong path"))?;
```

Comment:
Adding the path would be useful.

Reply:
Well ... it would ... But conversion of tmp dir's ```Path``` into a usable string is basically what fails here, so I cannot print it. May be I can rephrase it to something like "Temporary directory cannot be found"